### PR TITLE
[Hub Generated] Release machinelearningservices microsoft.machine learning services 2023 10 01

### PR DESCRIPTION
**Pull request by kseager**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/release-machinelearningservices-Microsoft.MachineLearningServices-2023-06-01-preview...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by TimLovellSmith** [comment]
>                 "minimum": 10

nit, I'd suggest don't document min/maximum here, just leave your server the flexibility to set its own policy

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/machineLearningServices.json:168 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>             "description": "Create or update request is successful.",

this is... not exactly the way I would want to describe it. 200 should always mean the resource was Updated.... without being created. I.e. it was effectively an 'update' request.
201 should mean it was created.

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/mfe.json:548 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         }

I wonder if the SDK codegen etc allows this multiple inheritance looking pattern?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/mfe.json:9470 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         }

I don't suppose you actually wanted to replace Datastore as the base type, or insert AzureDatastore in the inheritance sandwich instead?

---
In reply to: [1716631330](https://github.com/Azure/azure-rest-api-specs/pull/24949#issuecomment-1716631330) [](http://example.com/codeflow?ancestors=1716631330)

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/mfe.json:9470 in 3ec6f33. [](commit_id = 3ec6f33430feff940b64a261f3694ab33833c40c, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>         }

I'm curious if you need these properties - are they not redundant with the resource's id property?

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/mfe.json:9578 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/examples/Compute/restart.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "parameters": {
+    "subscriptionId": "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345",
+    "resourceGroupName": "testrg123",
+    "workspaceName": "workspaces123",
+    "computeName": "compute123",
+    "api-version": "2023-10-01"
+  },
+  "responses": {
+    "202": {
```
Comment (TimLovellSmith) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/examples/Compute/restart.json:10]: >": [](http://example.com/codeflow?start=8&length=2)

Doesn't HTTP say 202 Accepted should come with a Location header?

**Comment by TimLovellSmith** [comment]
>             "description": "Success."

Should probably say something more like 
"successfully started the long running operation. to get status..."

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/machineLearningServices.json:1304 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           },

Should probably say something more like successfully started the long running operation.
[noting  "x-ms-long-running-operation": true,]

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/machineLearningServices.json:1217 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           },

Should probably say something more like successfully started the long running operation.
[noting  "x-ms-long-running-operation": true,]

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/machineLearningServices.json:1305 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

**Comment by kseager** [comment]
all of the changes in this PR have been previously ARM approved. This PR is a batching of API changes so we can introduce the version in main with all the changes

**Comment by kseager** [comment]
PRs Currently Merged:
https://github.com/Azure/azure-rest-api-specs/pull/24968
https://github.com/Azure/azure-rest-api-specs/pull/25072
https://github.com/Azure/azure-rest-api-specs/pull/25289
https://github.com/Azure/azure-rest-api-specs/pull/25374
https://github.com/Azure/azure-rest-api-specs/pull/25519
https://github.com/Azure/azure-rest-api-specs/pull/25765
https://github.com/Azure/azure-rest-api-specs/pull/25775
https://github.com/Azure/azure-rest-api-specs/pull/25619

Coming soon/WIP Approval/Merge:
https://github.com/Azure/azure-rest-api-specs/pull/25801
https://github.com/Azure/azure-rest-api-specs/pull/25729
https://github.com/Azure/azure-rest-api-specs/pull/25594

**Comment by visingla-ms** [comment]
>         }

Please respond to this.

---
In reply to: [1716633059](https://github.com/Azure/azure-rest-api-specs/pull/24949#issuecomment-1716633059) [](http://example.com/codeflow?ancestors=1716633059)

---
Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/mfe.json:9578 in 5fb8294. [](commit_id = 5fb829494ad9388ad5baba80e9e74d79d377a163, deletion_comment = False)

**Comment by kseager** [comment]
> > ```
> >     }
> > ```
> 
> Please respond to this.
> 
> In reply to: [1716633059](https://github.com/Azure/azure-rest-api-specs/pull/24949#issuecomment-1716633059)
> 
> Refers to: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2023-10-01/mfe.json:9578 in [5fb8294](https://github.com/Azure/azure-rest-api-specs/commit/5fb829494ad9388ad5baba80e9e74d79d377a163). [](commit_id = [5fb8294](https://github.com/Azure/azure-rest-api-specs/commit/5fb829494ad9388ad5baba80e9e74d79d377a163), deletion_comment = False)

As I have mentioned before, all of these changes are either legacy or pre-reviewed/approved and are team specific. This PR is just to add them in as a unit.

**Comment by kseager** [comment]
> Please fix circular references as mentioned in other swagger PR also.

I have these this PR's which is approved and will be merged soon: https://github.com/Azure/azure-rest-api-specs/pull/25879
to fix this request

**Comment by kseager** [comment]
merge conflict https://github.com/Azure/azure-rest-api-specs/pull/26085

**Comment by ms-zhenhua** [comment]
please resolve the branch conflict and then run **/pr RequestMerg**e to request merge

**Comment by kseager** [comment]
will merge this PR through proxy/shadow: https://github.com/Azure/azure-rest-api-specs/pull/26104

**Pull request closed without merging**